### ISSUE TITLE
Misspelled `False` as keyword argument initialization not detected by rubocop

### ISSUE THREAD

**Issue by nadjaheitmann** [rubocop/rubocop#10296]
In this PR, there is a initialization of a keyword argument which is boolean. The `false` is spelled `False` which should throw an error in my opinion, but it did not.  

Link: https://github.com/theforeman/foreman_salt/pull/136/files#diff-179b69b42b6edeaca2c6102e7522527493004e93a18e24f539fb91d6cfdc04cdR142

--------

## Expected behavior
Give an error either to point out that the initialization value is misspelled or undefined.

## Actual behavior
Error was ignored. 

## Steps to reproduce the problem
Define keyword argument like `def myfunction`

## RuboCop version

On our ci:
```
bundle exec rubocop -V
0.89.1 (using Parser 3.0.3.2, rubocop-ast 0.8.0, running on ruby 2.5.9 x86_64-linux)
```
I verified with another test script and newer rubocop:
```
10:50 $ rubocop -V
1.22.0 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 2.7.0 x86_64-linux-gnu)
  - rubocop-minitest 0.10.3
  - rubocop-performance 1.8.1
  - rubocop-rails 2.8.1
```

**Comment by koic** [comment]
For example, it is not a misspelling for an application that has a user-defined `False` module. And RuboCop's static analysis cannot detect that false positive.

I think it's better to check spelling in your application using `Naming/InclusiveLanguage` configuration.

```yaml
Naming/InclusiveLanguage:
  Enabled: true
  FlaggedTerms:
    'False':
      Suggestions: 'false'
```

**Comment by nadjaheitmann** [comment]
Thanks for the quick reply, I will consider adding that to our configuration! I agree that marking `False/True` as errors leads to false positives. IMHO, I prefer to have a false warning that I can ignore/turn off rather than an error which is not found, though. But that is just personal preference.

**Comment by koic** [comment]
For example, `FALSE` deprecated constant removed in Ruby 3.0 can be detected by `Lint/DuplicateMethods` cop. However, `False` unlike `FALSE` does not exist in the language specification. This is my opinion as a maintainer that it's hard to detect typo (`NameError`) by default that isn't in the language spec. Thank you.
